### PR TITLE
Add a unit test for the network plugin.

### PR DESCRIPTION
**Pull request by octo**
This is not the best of tests; it was just something I found lying around in my workspace and thought I'd upload it here in case somebody finds it useful.

ChangeLog: network plugin: A unit test has been added.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -1370,6 +1370,14 @@ network_la_CPPFLAGS += $(GCRYPT_CPPFLAGS)
 network_la_LDFLAGS += $(GCRYPT_LDFLAGS)
 network_la_LIBADD += $(GCRYPT_LIBS)
 endif
+
+test_plugin_network_SOURCES = src/network_test.c \
+			      src/daemon/configfile.c \
+			      src/daemon/types_list.c
+test_plugin_network_CPPFLAGS = $(AM_CPPFLAGS)
```
Comment (rubenk) [inline Makefile.am]: This is not needed since you don't override the flags.

**Comment by rubenk** [comment]
Looks good, thanks! The build failure looks real though, can you resolve that?

File: Makefile.am (hunk 1 above)
Comment (octo) [reply Makefile.am]: Fixed, thanks!

File: Makefile.am [hunk 2]
```diff
@@ -1371,12 +1371,18 @@ network_la_LDFLAGS += $(GCRYPT_LDFLAGS)
...
-			      src/daemon/types_list.c
-test_plugin_network_CPPFLAGS = $(AM_CPPFLAGS)
-test_plugin_network_LDFLAGS = $(PLUGIN_LDFLAGS)
-test_plugin_network_LDADD = liboconfig.la libplugin_mock.la libmetadata.la
+test_plugin_network_SOURCES = \
+	src/network_test.c \
+	src/utils_fbhash.c \
+	src/daemon/configfile.c \
+	src/daemon/types_list.c
+test_plugin_network_LDFLAGS = $(PLUGIN_LDFLAGS) $(GCRYPT_LDFLAGS)
```
Comment (rubenk) [inline Makefile.am]: You'll need `test_plugin_network_CPPFLAGS = $(AM_CPPFLAGS) $(GCRYPT_CPPFLAGS)` as well I think?

Comment (octo) [reply Makefile.am]: Good catch, thanks!

**Comment by rpv-tomsk** [comment]
Heh, I will not merge this, even if I want.
If something goes wrong after merge master, I will not be able to solve that issue.

 #2958

**Comment by rpv-tomsk** [comment]
cc @mrunge 

>If something goes wrong after merge master, I will not be able to solve that issue.

That is really funny, I know:

```
Undefined			first referenced
 symbol  			    in file
inet_pton                           src/test_plugin_network-network_test.o  (symbol belongs to implicit dependency /lib/libnsl.so.1)
inet_ntop                           src/test_plugin_network-network_test.o  (symbol belongs to implicit dependency /lib/libnsl.so.1)
ld: fatal: symbol referencing errors. No output written to test_plugin_network
gmake[2]: *** [Makefile:6163: test_plugin_network] Error 2
```

Should @rjablonx  jump up and decide failed CI issue ?

**Comment by rpv-tomsk** [comment]
I wish you good luck.

**Comment by rpv-tomsk** [comment]
Ok, when I was "trusted contributor" I made PR exactly to fix CI problems.
I do that mostly because I can do PR, merge it without any delay, check for master branch CI results, apply new fixes and merge these fixes without any delay again.

For now, if any "untrusted contributor" will try to solve such issue, he should publish PR and sit and wait for somebody notice. Maybe stars come together and somebody pay attention on it to merge it to master.

When CI will fail again, the cycle (with mandatory sit-and-wait stage) will have to repeat again.

Good luck.

**Comment by rpv-tomsk** [comment]
cc @faxm0dem @sunkuranganath

**Comment by sunkuranganath** [comment]
@rpv-tomsk you have a unique way of communicating the issues :D 

I believe issues like these are not uncommon in widely used open source communities. However the resolution of issues like rights, CI, permissions, requests, etc., gets resolved fairly soon due to good communication.
We all work on this in our own time on top of our other responsibilities and want to make this work. 
As @mrunge describes in #2958 once we get to fix the issues I hope things will get a bit easier.

Please note that @mrunge @tokkee and myself are working on a Collectd release proposal that touches upon these things and get some direction/method to this madness. Will send out the draft in next few days as we are almost done with our updates. Hope to get your feedback to make this easy for you and rest of the community and establish some sense.

**Comment by rpv-tomsk** [comment]
>We all work on this in our own time on top of our other responsibilities and want to make this work.

I'm speaking about the same - our's time is the most valuable resource, so we should not spent it on fighting administrative and CI issues.  Also, I think, nobody of us wants to spent time implementing feature and then throw out the result of the work just because lack of response. Yes, I know that is the problem 'code owners' tries to solve (add plugin owners ability to merge theirs changes), but that breaks things and puts them upside down.

**Comment by rpv-tomsk** [comment]
Again: When I merge somebody's PR, I'm responsible for that merge. That is idea and goal of review process.

But in current situation, I _have no permission_ to fix issues, caused by that merge. 
So why do you ask me to be "(un)trusted maintainer"?

Another cause of this PR problem is different set of CI checks applied to 'master' and 'PR' branches.
Even if 'PR' branch check succeded, 'master' branch check can fail, which is what happened.

All these are administrative issues, which needs to be solved, but lasts for months and years.

**Pull request merged**